### PR TITLE
Update kilonova.py

### PR DESCRIPTION
**Pull request by AkshayPanayada**

### CODE REVIEW

File: skysurvey/target/kilonova.py [hunk 1]
```diff
@@ -1,12 +1,32 @@
 from .core import Transient
-
+#This is the code to model kilonova
+import sncosmo as sc
+from simsurvey.utils import model_tools
+from simsurvey.models import AngularTimeSeriesSource
+import numpy as np
+optional_injection_parameters={}
+phase, wave, cos_theta, flux = model_tools.read_possis_file(
```
Comment (mcoughlin) [inline skysurvey/target/kilonova.py:8]: simsurvey should not be a dependency of skysurvey. You should copy in what code you need to read this file. You will also need to add the file into the repo in the same pull request.

**Review by mcoughlin — CHANGES_REQUESTED** [review]
You will need to copy in the code from simsurvey you need here.

File: skysurvey/target/kilonova.py (hunk 1 above)
Comment (AkshayPanayada) [reply skysurvey/target/kilonova.py:8]: ok will do that

File: skysurvey/target/kilonova.py [hunk 2]
```diff
@@ -1,12 +1,161 @@
...
+#This is the code to model kilonova
+import sncosmo as sc
+from scipy.interpolate import (InterpolatedUnivariateSpline as Spline1d,
+                               RectBivariateSpline as Spline2d)
+import numpy as np
+optional_injection_parameters={}
+
+def read_possis_file(filename):
+    """Read in a spectral model created by POSSIS (1906.04205), as appropriate
+       for injestion as a simsurvey.models.AngularTimeSeriesSource. Model
```
Comment (mcoughlin) [inline skysurvey/target/kilonova.py:12]: this docstring needs to be updated now that it is skysurvey.

File: skysurvey/target/kilonova.py [hunk 3]
```diff
@@ -1,12 +1,161 @@
...
+        if self._zero_before:
+            mask = np.atleast_1d(phase) < self.minphase()
+            f[mask, :] = 0.
+        if self._zero_after:
+            mask = np.atleast_1d(phase) > self.maxphase()
+            f[mask, :] = 0.
+        return f
+
+phase, wave, cos_theta, flux = model_tools.read_possis_file(
+                "nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt")
```
Comment (mcoughlin) [inline skysurvey/target/kilonova.py:138]: You also need to commit this file somewhere in the repo. Maybe in a data/ folder?

**Comment by AkshayPanayada** [comment]
I need to do a bit more work to make it easy for the user. Closing all the pull requests to change the kilonova.py file

**Pull request closed without merging**